client/blocks/cookie-banner/index.tsx

```diff
@@ -0,0 +1,69 @@
+import { CookieBanner, CookieBannerProps } from '@automattic/privacy-toolset';
+import cookie from 'cookie';
+import { noop } from 'lodash';
+import { useCallback, useEffect, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import {
+	refreshCountryCodeCookieGdpr,
+	setTrackingPrefs,
+	shouldSeeCookieBanner,
+	getTrackingPrefs,
+} from 'calypso/lib/analytics/utils';
+import { bumpStat } from 'calypso/state/analytics/actions';
+import { useCookieBannerContent } from './use-cookie-banner-content';
+
+export const isServer = typeof document === 'undefined';
+
+const CookieBannerInner = ( { onClose }: { onClose: () => void } ) => {
+	const content = useCookieBannerContent();
+	const dispatch = useDispatch();
+
+	const handleAccept = useCallback< CookieBannerProps[ 'onAccept' ] >(
+		( buckets ) => {
+			setTrackingPrefs( { ok: true, buckets } );
+			onClose();
+		},
+		[ onClose ]
+	);
+
+	useEffect( () => {
+		dispatch(
+			bumpStat(
+				'cookie-banner-view',
+				'total,' + document.location.host.replace( /[^a-zA-Z0-9]/g, '-' )
+			)
+		);
+	}, [ dispatch ] );
+
+	return <CookieBanner content={ content } onAccept={ handleAccept } />;
+};
+
+const CookieBannerContainer = () => {
+	const [ show, setShow ] = useState( false );
+
+	useEffect( () => {
+		refreshCountryCodeCookieGdpr().then( () => {
```

Comment:
Should we be effectively canceling the corresponding requests in the event of unmounting this component?

Reply:
As far as I know, it's impossible to cancel js promises. Or do you have something else in mind?

The only thing that comes to my mind is to prevent execution of code in `then` scope when the component is unmounted - and I think that's a good idea.

Reply:
FYI, [`AbortController`](https://developer.mozilla.org/en-US/docs/Web/API/AbortController) allows for a cancellable promise approach in pure JS, and the native browser `fetch` supports cancelling requests through that mechanism.

Reply:
Thanks, I wasn't aware of that, will adjust accordingly.